FWCore/Framework/src/globalTransitionAsync.h

```diff
@@ -113,6 +136,27 @@ namespace edm {
     iSchedule.processOneGlobalAsync<Traits>(std::move(h), iPrincipal, iES, token);
   }
 
+  // The only purpose of this function is to create dummy values for the
+  // EventSetup arguments needed by the generic function beginGlobalTransitionsAsync
+  template <typename Traits, typename P, typename SC>
+  void beginGlobalTransitionAsync(
+      WaitingTaskHolder iWait, Schedule& iSchedule, P& iPrincipal, ServiceToken const& token, SC& iSubProcesses) {
+    // There is no EventSetup for ProcessBlock transitions
+    // so we just pass in dummy values that are not used.
+    IOVSyncValue dummyIOVSyncValue;
+    EventSetupImpl dummyEventSetupImpl;
```

Comment:
I'm really not a fan of this. Should we redo the `beginGlobalTransitionAsync` to take as argument something form the `Traits` which can be used to deal with EventSetup related information?

Reply:
I spent a while improving the interface in globalTransitionAsync.h. I think it is better now. There were many ways this could be done. Let me know if you have comments and want me to do this differently.  The changes are in a separate commit (currently the last one).

After this there is one more comment I still need to address, so I am not starting the tests yet.